### DISCUSSION TITLE
Allow git send-email to create PR's

### DISCUSSION THREAD

**Discussion by williewillus** [community/community#9082] [Pull Requests]
My absolute least favorite part of working with Github is the need to perform a laborious fork, clone the fork, branch, push, then create PR steps for any repo I want to contribute to. Often I spot small issues in projects I use, and would like to do a drive by spot-fix...but then I think of having to do the five steps above (six if you count having to delete the useless fork I just made after my change is accepted), and that quickly saps my motivation.

It would be incredibly amazing if GitHub supported mailing patches (via `git send-email`) to a repository, e.g. `github/feedback@patches.github.com` which would automatically generate and open a pull request on the patch author's behalf with the mailed commits applied to the base branch.

Contributing to a repository then would be shortened to three steps: clone upstream repo, branch (optional), git send-email.

Commits would have to be signed with a PGP key associated with the account of the author's email address to prevent spoofing, and invalid signatures or mail to private, nonexistent, or otherwise unauthorized repositories would be bounced.

I have a sensation that GitHub likes to encourage the "fork everything" workflow because it pads the numbers for repositories forked on the website, but please consider the actual workflow of many people who have tried the alternative systems (mailing list patches, Phabricator, etc.) and found them much, much smoother of a workflow. 

Thanks.

**Comment by Tachi107** [comment]
This has been already suggested a while back in https://github.com/isaacs/github/issues/1423
